### ISSUE TITLE
9.0.0: Historic code migrations no longer validate

### ISSUE THREAD

**Issue by henricook** [flyway/flyway#3481]
###### Which version and edition of Flyway are you using?

Testing upgrade from 8.5.13 to 9.0.0 - community

###### Which client are you using? (Command-line, Java API, Maven plugin, Gradle plugin)
Java API

###### Which database are you using? (Type & version)
Postgres 10.6

###### Which operating system are you using?
Linux

###### What did you do? (Please include the content causing the issue, any relevant configuration settings, the SQL statement(s) that failed (if any), and the command you ran)
When starting our application with version 9.0.0 of flyway on a database made by 8.5.13 of flyway the following errors appear for historic 'code migrations' that we have performed (most of our migrations are simple .sql files)

Our current version number is somewhere in the 40s - so these are very old major version code migrations

###### What did you expect to see?
I expected to see no migration issues for old code migrations

###### What did you see instead?

```
org.flywaydb.core.api.exception.FlywayValidateException: Validate failed: Migrations have failed validation
Migration checksum mismatch for migration version 5
-> Applied to database : 0
-> Resolved locally    : null
Either revert the changes to the migration, or run repair to update the schema history.
Migration checksum mismatch for migration version 6
-> Applied to database : 0
-> Resolved locally    : null
Either revert the changes to the migration, or run repair to update the schema history.
Migration checksum mismatch for migration version 7
-> Applied to database : 0
-> Resolved locally    : null
Either revert the changes to the migration, or run repair to update the schema history.
```

**Comment by henricook** [comment]
So I think the core of this is... that code migration checksums are now `null` instead of `0`?

**Comment by DoodleBobBuffPants** [comment]
Thanks for reporting this, this is fixed by 6bc712ce9d08c77e50d8de40bd04e0e5a08740df and there will be a release later today
